src/winutils.cc

```diff
@@ -0,0 +1,126 @@
+
+
+#include "winutils.h"
+#include "TlHelp32.h"
+#include <cassert>
+
+
+DWORD getppid() {
+    HANDLE hSnapshot = INVALID_HANDLE_VALUE;
+    PROCESSENTRY32 pe32;
+    DWORD ppid = 0, pid = GetCurrentProcessId();
+
+    hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
```

Comment:
yes, I think so. It should work. I had taken the original code from msdn where they were just looking at one process. Will see if this (without using pid=0) works.